### PR TITLE
Marine/feature/event pictures

### PR DESCRIPTION
**Pull request by Marsobad**
Features to upload pictures for events, similar to upload images for animals
- I added a new data model : EventPicture
- I added the connection with the petEvent table
- new controller

The only difference is that you may upload a picture for an event even if you did not attend this event (like on Facebook for instance). Yet, if you did not attend the event you cannot delete pictures.

### CODE REVIEW

**Comment by Marsobad** [comment]
> I think it would be a good idea to keep track of which user has uploaded which picture. In that case, everyone can upload picture for an event and only the creator of the event & the uploader of the picture can delete it.

Yeah, I think so too so I had added a field userId to eventPictures model but I needed to fix some stuffs before commiting

File: back/src/database/models/event.ts [hunk 1]
```diff
@@ -54,15 +55,23 @@ export class PetEvent extends Model {
...
     public readonly authorizedSpecies?: Specie[];
     public readonly comments?: EventComment[];
+    public readonly eventPictures?:EventPicture[];
 
     public static associations: {
         attendees: Association<PetEvent, Animal>;
         authorizedSpecies: Association<PetEvent, Specie>;
-        comments: Association<PetEvent, EventComment>
+        comments: Association<PetEvent, EventComment>;
+        eventPictures: Association<EventPicture, EventComment>;
```
Comment (Hackatosh) [inline back/src/database/models/event.ts]: Swap EventPicture and EventComment for type definition

File: back/src/database/models/eventPicture.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+/***
+ * Model used to represent a picture related to an event in the DB.
+ * The filename is used to interact with the SFTP server.
+ * The contentType is used to provide correct headers when sending the picture through HTTP response.
+ ***/
+
+export class EventPicture extends Model {
+    public id!: number;
+    public eventId !: number;
+    public userId: number;
```
Comment (Hackatosh) [inline back/src/database/models/eventPicture.ts]: Add a !

File: back/src/routes/controllers/animalPictures.ts [hunk 3]
```diff
@@ -123,16 +123,18 @@ animalPicturesRouter.delete('/', deleteAnimalPictureChecks, inputValidationMW, a
         res.status(404).json({message: "This animal does not exist"});
         return;
```
Comment (Hackatosh) [inline back/src/routes/controllers/animalPictures.ts]: useless return

File: back/src/routes/controllers/animalPictures.ts [hunk 4]
```diff
@@ -123,16 +123,18 @@ animalPicturesRouter.delete('/', deleteAnimalPictureChecks, inputValidationMW, a
...
     }
-    if (pet.userId !== userId) {
+    else if (pet.userId !== userId) {
         res.status(403).json({message: "You don't have access to this animal"});
         return;
```
Comment (Hackatosh) [inline back/src/routes/controllers/animalPictures.ts]: useless return

File: back/src/routes/controllers/eventPictures.ts [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+eventPicturesRouter.delete('/', deleteEventPictureChecks, inputValidationMW, async (req: AuthenticatedRequest, res: Response) => {
+    const filename = req.query.filename;
+    const userId = req.authInfos.userId;
+    const eventId = req.params.eventId;
+    const file = await EventPicture.findOne({where: {filename: filename}});
+    if (!file) {
+        res.status(404).json({message: "This file does not exist."});
+        return;
+    }
+    const picture = await EventPicture.findOne({where: {id: file.eventId}});
```
Comment (Hackatosh) [inline back/src/routes/controllers/eventPictures.ts]: Wrong : you're trying to obtain the file defined just before. 
This should be deleted.

File: back/src/routes/controllers/eventPictures.ts [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+    const event = await PetEvent.findOne({where:{id: file.eventId}});
+
+    if (!picture) {
+        res.status(404).json({message: "This event does not exist"});
+        return;
+    }
+
+
+    else if (picture.userId !== userId && event.userId !== userId ) {
```
Comment (Hackatosh) [inline back/src/routes/controllers/eventPictures.ts]: Should be file instead of picture

File: back/src/routes/controllers/eventPictures.ts [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+        res.status(403).json({message: "You don't have access to this picture "});
+        return;
```
Comment (Hackatosh) [inline back/src/routes/controllers/eventPictures.ts]: useless return

File: back/src/routes/controllers/eventPictures.ts [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+    const filename = req.query.filename;
+    const userId = req.authInfos.userId;
+    const eventId = req.params.eventId;
+    const file = await EventPicture.findOne({where: {filename: filename}});
+    if (!file) {
+        res.status(404).json({message: "This file does not exist."});
+        return;
+    }
+    const picture = await EventPicture.findOne({where: {id: file.eventId}});
+    const event = await PetEvent.findOne({where:{id: file.eventId}});
```
Comment (Hackatosh) [inline back/src/routes/controllers/eventPictures.ts]: What if event is null ?

File: back/src/routes/controllers/eventPictures.ts [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+
+    if (!picture) {
```
Comment (Hackatosh) [inline back/src/routes/controllers/eventPictures.ts]: Same as file, should be deleted

File: back/src/routes/controllers/animalPictures.ts (hunk 3 above)
Comment (Marsobad) [reply back/src/routes/controllers/animalPictures.ts]: ok, but why?

**Pull request merged**